### PR TITLE
Add some functions for User and agreement management

### PR DESCRIPTION
**Pull request by RobFaie**

### CODE REVIEW

File: PwshAdobeSign.psd1 [hunk 1]
```diff
@@ -70,10 +70,21 @@
...
+        'Get-Connection',                'Get-AdobeSignConnection'
+        'Get-Group',                     'Get-Group'
+        'Get-User',                      'Get-User'
+        'Get-UserGroup',                 'Get-UserGroup'
+        'Hide-Agreement',                'Hide-Agreement'
+        'Invoke-Method',                 'Invoke-AdobeSignMethod'
+        'Remove-UserGroup',              'Remove-UserGroup'
+        'Show-Agreement',                'Show-Agreement'
+        'Update-AgreementVisibility',    'Update-AgreementVisibility'
+        'Update-UserGroup',              'Update-UserGroup'
```
Comment (mderriey) [inline PwshAdobeSign.psd1:87]:
```suggestion
        'Add-UserGroup',                 'Add-AdobeSignUserGroup'
        'Connect-',                      'Connect-AdobeSign'
        'Get-Agreement',                 'Get-AdobeSignAgreement'
        'Get-AgreementCombinedDocument', 'Get-AdobeSignAgreementCombinedDocument'
        'Get-BaseUri',                   'Get-AdobeSignBaseUri'
        'Get-Connection',                'Get-AdobeSignConnection'
        'Get-Group',                     'Get-AdobeSignGroup'
        'Get-User',                      'Get-AdobeSignUser'
        'Get-UserGroup',                 'Get-AdobeSignUserGroup'
        'Hide-Agreement',                'Hide-AdobeSignAgreement'
        'Invoke-Method',                 'Invoke-AdobeSignMethod'
        'Remove-UserGroup',              'Remove-AdobeSignUserGroup'
        'Show-Agreement',                'Show-AdobeSignAgreement'
        'Update-AgreementVisibility',    'Update-AdobeSignAgreementVisibility'
        'Update-UserGroup',              'Update-AdobeSignUserGroup'
```

File: functions/Add-UserGroup.ps1 [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+        Adds a user to a group.
+    .EXAMPLE
+        PS /> Add-AdobeSignUserGroup -Context $context -UserId hXy4R2NaYnvTaftrEhaD4ZAJrxh3YM8kuf8CupEouFoK -GroupId hXy4R2NaYnvTaftrEhaD4ZAJrxh3YM8kuf8CupEouFoK
+
+        Adds a user to a group using a connection context.
+    #>
+    [OutputType([PSCustomObject])]
+    [CMDletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    Param (
```
Comment (mderriey) [inline functions/Add-UserGroup.ps1:19]: `UserId` and `GroupId` are declared as integers, but the example uses strings.

File: functions/Get-Agreement.ps1 [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    .EXAMPLE
+        PS /> Get-AdobeSignAgreement -Context $context
+
+        Gets a list of all agreements using a connection context.
+    #>
+    [OutputType([PSCustomObject])]
+    [CMDletBinding(DefaultParameterSetName = 'all')]
+    Param (
+        # The ID of the agreement.
+        [Parameter(ParameterSetName = 'id')]
```
Comment (mderriey) [inline functions/Get-Agreement.ps1:33]: Would it need to be mandatory given it's the only parameter for this set?

File: functions/Get-Agreement.ps1 [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    .SYNOPSIS
+        Returns a list of agreements or a specified agreement.
+    .DESCRIPTION
+        Returns a list of agreements or a specified agreement.
+    .EXAMPLE
+        PS /> Get-AdobeSignAgreement
+
+        Gets a list of all agreements
+    .EXAMPLE
+        PS /> Get-AdobeSignAgreement -Id hXy4R2NaYnvTaftrEhaD4ZAJrxh3YM8kuf8CupEouFoK
```
Comment (mderriey) [inline functions/Get-Agreement.ps1:13]: Same comment about mismatch between parameter types as declared and in the example.

**Review by mderriey — APPROVED** [review]

File: functions/Add-UserGroup.ps1 (hunk 2 above)
Comment (RobFaie) [reply functions/Add-UserGroup.ps1:19]: updated

File: functions/Get-Agreement.ps1 (hunk 4 above)
Comment (RobFaie) [reply functions/Get-Agreement.ps1:13]: updated

File: functions/Get-Agreement.ps1 (hunk 3 above)
Comment (RobFaie) [reply functions/Get-Agreement.ps1]: Yes it would.

**Pull request merged**